### PR TITLE
cli: add --geo-libs flag for geospatial library location

### PR DESCRIPTION
**Pull request by otan**
Add a flag to the CLI (--geo-libs) which contains the libraries with
geospatial capabilities we need (e.g. GEOS, maybe PROJ4/SFCGAL later).
This is used by GEOS in addition to looking at parenting directories.

Also log the directory which GEOS was initialized.

Release note: None

### CODE REVIEW

**Review by petermattis — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47759)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @petermattis)

___
*[pkg/geo/geos/geos_unix.go, line 87 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-M5SJXaEBWC08EFx_KW4:-M5SJXaEBWC08EFx_KW5:b-pwd9mc) ([raw file](https://github.com/cockroachdb/cockroach/blob/8124961cde1a51bb9531a0159f04e210807556bb/pkg/geo/geos/geos_unix.go#L87)):*
> ```Go
> 			"/usr/lib",
> 			"/lib",
> 			"/usr/local/lib",
> ```

Do we want to look in system directories for `libgeos_c`? My expectation is that we'll be shipping `libgeos_c` along with the Cockroach binary.

**Comment by otan** [comment]
Not sure the failures are me.

**Review by petermattis — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47759)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/geo/geos/geos_unix.go, line 87 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-M5SJXaEBWC08EFx_KW4:-M5SR1dd70D3MHFhvkSh:b21lhhc) ([raw file](https://github.com/cockroachdb/cockroach/blob/8124961cde1a51bb9531a0159f04e210807556bb/pkg/geo/geos/geos_unix.go#L87)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

(happy to make it just that)
</blockquote></details>

I don't know if there is a standard place for installation. On `roachprod` we just place the binary in the home directory. In k8s it is run out of a container. Perhaps check with the docs folks to see if we make a recommendation anywhere.

**Comment by otan** [comment]
___
*[pkg/geo/geos/geos_unix.go, line 87 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-M5SJXaEBWC08EFx_KW4:-M5T_TUB-sewiwW7usds:b-hz9xkh) ([raw file](https://github.com/cockroachdb/cockroach/blob/8124961cde1a51bb9531a0159f04e210807556bb/pkg/geo/geos/geos_unix.go#L87)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

I don't know if there is a standard place for installation. On `roachprod` we just place the binary in the home directory. In k8s it is run out of a container. Perhaps check with the docs folks to see if we make a recommendation anywhere. 
</blockquote></details>

I think I've updated it to the reasonable /usr based once.
We can update this list later -- but the flag is the main bit we should include :D

**Review by petermattis — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47759)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @rmloveland)

___
*[pkg/cli/cliflags/flags.go, line 1054 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-M5TeQbQ9sCLYTbBIftO:-M5TeQbRDp2hInkRoccN:b-48z0m6) ([raw file](https://github.com/cockroachdb/cockroach/blob/620a517e6ca554652459907f7983b17ca7a734d5/pkg/cli/cliflags/flags.go#L1054)):*
> ```Go
> 
> 	GEOSLocation = FlagInfo{
> 		Name: "geos",
> ```

Is `geos` the only library we'll be loading like this? I wonder if this should be a more general `libs` flag.
___
*[pkg/geo/geos/geos_unix.go, line 87 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-M5SJXaEBWC08EFx_KW4:-M5TfhIC6Fbc-3hnuz4_:b-d7grw6) ([raw file](https://github.com/cockroachdb/cockroach/blob/8124961cde1a51bb9531a0159f04e210807556bb/pkg/geo/geos/geos_unix.go#L87)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

I think I've updated it to the reasonable /usr based once.
We can update this list later -- but the flag is the main bit we should include :D 
</blockquote></details>

I'm going to push back on any system directories again. What if we accidentally find a `libgeos_c` in one of these default directories and the version is incompatible with our needs? I think we only want to use the version of geos that ships with CRDB. The question is where that will be installed. Perhaps @rmloveland has some thoughts about this.

File: pkg/cli/flags.go [hunk 1]
```diff
@@ -363,6 +363,7 @@ func init() {
 		StringFlag(f, &startCtx.listeningURLFile, cliflags.ListeningURLFile, startCtx.listeningURLFile)
 
 		StringFlag(f, &startCtx.pidFile, cliflags.PIDFile, startCtx.pidFile)
+		StringFlag(f, &startCtx.geosLocation, cliflags.GEOSLocation, "")
```
Comment (knz) [inline pkg/cli/flags.go:366]:
```go
StringFlag(f, &startCtx.geosLocation, cliflags.GEOSLocation, startCtx.geosLocation)
```

then initialize `startCtx.geosLocation` in `context.go` in the same directory. Thanks

File: pkg/cli/flags.go [hunk 2]
```diff
@@ -657,6 +658,7 @@ func init() {
 	// The --empty flag is only valid for the top level demo command,
 	// so we use the regular flag set.
 	BoolFlag(demoCmd.Flags(), &demoCtx.useEmptyDatabase, cliflags.UseEmptyDatabase, demoCtx.useEmptyDatabase)
+	StringFlag(demoFlags, &demoCtx.geosLocation, cliflags.GEOSLocation, "")
```
Comment (knz) [inline pkg/cli/flags.go:661]: ditto (Although maybe you could share the same field `startCtx.geosLocation` for both commands.)

File: pkg/geo/geos/geos_unix.go [hunk 3]
```diff
@@ -44,29 +44,77 @@ var geosOnce struct {
...
+			"/usr/local/lib",
+			"/usr/lib/x86_64-linux-gnu",
+			"/usr/lib/i386-linux-gnu",
+			"/usr/lib",
+		}
+	}
+	var locs []string
+	locs = append(locs, findGEOSLocationsInParentingDirectories(ext)...)
+	for _, loc := range locDirs {
+		locs = append(locs, filepath.Join(loc, "libgeos_c."+ext))
```
Comment (knz) [inline pkg/geo/geos/geos_unix.go:108]: Please log the directory that was selected when auto-detection succeeds. This is necessary for troubleshooting.

**Review by rmloveland — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47759)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @petermattis)

___
*[pkg/geo/geos/geos_unix.go, line 87 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-M5SJXaEBWC08EFx_KW4:-M5XXoB0C1A_81k8SA6w:b-ai8ffk) ([raw file](https://github.com/cockroachdb/cockroach/blob/8124961cde1a51bb9531a0159f04e210807556bb/pkg/geo/geos/geos_unix.go#L87)):*
tl;dr: I actually think we should *only* use the libgeos in /usr/local/lib, and if there's one already there, we should tell users to uninstall it and replace it with our version.  Every other solution is more complicated.

Longer version:

I just did a `brew install geos` and it does go into `/usr/local/lib` (it also requires OpenSSL 1.1, strangely).  This is IMO the only right place to put user-installed libraries on Unixes.  This is basically what they say in [the Linux File Hierarchy Standard (FHS)](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch04s09.html):

> The /usr/local hierarchy is for use by the system administrator when installing software locally. It needs to be safe from being overwritten when the system software is updated. It may be used for programs and data that are shareable amongst a group of hosts, but not found in /usr. Locally installed software must be placed within /usr/local rather than /usr unless it is being installed to replace or upgrade software in /usr. [28]

Regarding Peter's comment:

> What if we accidentally find a libgeos_c in one of these default directories and the version is incompatible with our needs? I think we only want to use the version of geos that ships with CRDB. 

I completely agree that this is the main problem.  From my POV as a user, it's much easier to fix "library not found in /usr/local/lib" than "we search 10 paths and picked up this wrong version of the library that causes subtle brokenness".  IMO we should have one and only one documented place to put our desired libgeos on Unixes: `/usr/local/lib`

It's much easier to explain "you must install libgeos *here* to work with CockroachDB" than to explain "we search these semi-random directories for a libgeos, which may or may not end up being the right one, and you will have to debug it if not"

**Comment by knz** [comment]
>  one and only one documented place to put our desired libgeos on Unixes: /usr/local/lib

Either that or the one configured via the command-line flag. I want to be able to override if I'm installing in my user account.

**Comment by rmloveland** [comment]
> Either that or the one configured via the command-line flag. I want to be able to override if I'm installing in my user account.

Yes, agreed - as with `configure --prefix`

File: pkg/geo/geos/geos_unix.go (hunk 3 above)
Comment (otan) [reply pkg/geo/geos/geos_unix.go:108]: i'll keep them, separate for now!

**Review by otan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47759)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz and @petermattis)

___
*[pkg/cli/flags.go, line 366 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-M5SsMy_8UhdQPWb0XeV-r2-366:-M5Xrvsr8JRZB1olfVKA:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/620a517e6ca554652459907f7983b17ca7a734d5/pkg/cli/flags.go#L366)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

```go
StringFlag(f, &startCtx.geosLocation, cliflags.GEOSLocation, startCtx.geosLocation)
```

then initialize `startCtx.geosLocation` in `context.go` in the same directory. Thanks
</blockquote></details>

Done.
___
*[pkg/cli/flags.go, line 661 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-M5SsMy_8UhdQPWb0XeV-r2-661:-M5Xrw7p4UT1ZP-UtmHN:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/620a517e6ca554652459907f7983b17ca7a734d5/pkg/cli/flags.go#L661)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

ditto (Although maybe you could share the same field `startCtx.geosLocation` for both commands.)
</blockquote></details>

Done.
___
*[pkg/cli/cliflags/flags.go, line 1054 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-M5TeQbQ9sCLYTbBIftO:-M5XrxXq9qYrBTWNNI41:b-n2ccha) ([raw file](https://github.com/cockroachdb/cockroach/blob/620a517e6ca554652459907f7983b17ca7a734d5/pkg/cli/cliflags/flags.go#L1054)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

Is `geos` the only library we'll be loading like this? I wonder if this should be a more general `libs` flag.
</blockquote></details>

for now. happy to make changes if we decide on more (3D, PROJ).
___
*[pkg/geo/geos/geos_unix.go, line 87 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-M5SJXaEBWC08EFx_KW4:-M5XrntB27PDhbZKM6kW:btahtzx) ([raw file](https://github.com/cockroachdb/cockroach/blob/8124961cde1a51bb9531a0159f04e210807556bb/pkg/geo/geos/geos_unix.go#L87)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

tl;dr: I actually think we should *only* use the libgeos in /usr/local/lib, and if there's one already there, we should tell users to uninstall it and replace it with our version.  Every other solution is more complicated.

Longer version:

I just did a `brew install geos` and it does go into `/usr/local/lib` (it also requires OpenSSL 1.1, strangely).  This is IMO the only right place to put user-installed libraries on Unixes.  This is basically what they say in [the Linux File Hierarchy Standard (FHS)](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch04s09.html):

> The /usr/local hierarchy is for use by the system administrator when installing software locally. It needs to be safe from being overwritten when the system software is updated. It may be used for programs and data that are shareable amongst a group of hosts, but not found in /usr. Locally installed software must be placed within /usr/local rather than /usr unless it is being installed to replace or upgrade software in /usr. [28]

Regarding Peter's comment:

> What if we accidentally find a libgeos_c in one of these default directories and the version is incompatible with our needs? I think we only want to use the version of geos that ships with CRDB. 

I completely agree that this is the main problem.  From my POV as a user, it's much easier to fix "library not found in /usr/local/lib" than "we search 10 paths and picked up this wrong version of the library that causes subtle brokenness".  IMO we should have one and only one documented place to put our desired libgeos on Unixes: `/usr/local/lib`

It's much easier to explain "you must install libgeos *here* to work with CockroachDB" than to explain "we search these semi-random directories for a libgeos, which may or may not end up being the right one, and you will have to debug it if not"
</blockquote></details>

sounds good to me.
___
*[pkg/geo/geos/geos_unix.go, line 108 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-M5SH81uwrmFMEoqB8DY-r2-108:-M5XruZe0X0YAs92zvH5:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/620a517e6ca554652459907f7983b17ca7a734d5/pkg/geo/geos/geos_unix.go#L108)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

i'll keep them, separate for now!
</blockquote></details>

Done.

**Review by petermattis — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47759)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz, @otan, and @rmloveland)

___
*[pkg/cli/cliflags/flags.go, line 1054 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-M5TeQbQ9sCLYTbBIftO:-M5XtdP_-CpSUHZI32Xb:b-sdnfpw) ([raw file](https://github.com/cockroachdb/cockroach/blob/620a517e6ca554652459907f7983b17ca7a734d5/pkg/cli/cliflags/flags.go#L1054)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

for now. happy to make changes if we decide on more (3D, PROJ).
</blockquote></details>

We don't have to decide in this PR, but should get this figured out before 20.2 ships as it will be more awkward to change at that point.
___
*[pkg/geo/geos/geos_unix.go, line 87 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-M5SJXaEBWC08EFx_KW4:-M5Xu17w12-wFm01VGBJ:b-i9zvj2) ([raw file](https://github.com/cockroachdb/cockroach/blob/8124961cde1a51bb9531a0159f04e210807556bb/pkg/geo/geos/geos_unix.go#L87)):*
> tl;dr: I actually think we should only use the libgeos in /usr/local/lib, and if there's one already there, we should tell users to uninstall it and replace it with our version. 

Do we have an "install time" at which to detect the presence of libgeos in /usr/local/lib? Honest question. I'm not familiar with our docs on this topic and only ever run cockroach out of the source directory or via `roachprod`.

> I completely agree that this is the main problem. From my POV as a user, it's much easier to fix "library not found in /usr/local/lib" than "we search 10 paths and picked up this wrong version of the library that causes subtle brokenness". IMO we should have one and only one documented place to put our desired libgeos on Unixes: /usr/local/lib

Why does `/usr/local/lib` get special treatment here? Let me propose an alternative: we don't search any default directories but require that the `--geos` flag be specified. For `brew`, `/usr/local/bin/cockroach` would become a shell script which would invoke `cockroach --geos=/usr/local/lib/libgeos_c.so`. The docker container would automatically have the correct `--geos` flag baked in. If there are any other packages of cockroach, they could take the shell script wrapper approach as well. (Are there any other packages of cockroach?)

**Review by otan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47759)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz and @petermattis)

___
*[pkg/cli/cliflags/flags.go, line 1054 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-M5TeQbQ9sCLYTbBIftO:-M5YAAMM6BZH4rkd8kkX:b-262rcq) ([raw file](https://github.com/cockroachdb/cockroach/blob/620a517e6ca554652459907f7983b17ca7a734d5/pkg/cli/cliflags/flags.go#L1054)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

We don't have to decide in this PR, but should get this figured out before 20.2 ships as it will be more awkward to change at that point.
</blockquote></details>

hmm, change of heart - let's have the discussion now. are we proposing something like`--geo_libs=/dir1,/dir2/....` which we specify each directory which we expect all libraries to be in, or exact files? does the `--geo_libs` flag make sense as a name?
___
*[pkg/geo/geos/geos_unix.go, line 87 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-M5SJXaEBWC08EFx_KW4:-M5YAavD7MBwtDvyZpvt:b-s2thzi) ([raw file](https://github.com/cockroachdb/cockroach/blob/8124961cde1a51bb9531a0159f04e210807556bb/pkg/geo/geos/geos_unix.go#L87)):*
https://www.cockroachlabs.com/docs/stable/install-cockroachdb-mac.html
https://www.cockroachlabs.com/docs/stable/install-cockroachdb-linux.html

We tell the users to put it straight into /usr/local/bin.

> Why does /usr/local/lib get special treatment here? 

factory default for everything I guess. I'm happy to omit having `/usr/local/lib` as the default for now and we can have that discussion next meeting (if we can't decide in person).

**Review by petermattis — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47759)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz, @otan, and @rmloveland)

___
*[pkg/cli/cliflags/flags.go, line 1054 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-M5TeQbQ9sCLYTbBIftO:-M5YUydU42T46mZgASl7:b-r3d8fw) ([raw file](https://github.com/cockroachdb/cockroach/blob/620a517e6ca554652459907f7983b17ca7a734d5/pkg/cli/cliflags/flags.go#L1054)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

hmm, change of heart - let's have the discussion now. are we proposing something like`--geo_libs=/dir1,/dir2/....` which we specify each directory which we expect all libraries to be in, or exact files? does the `--geo_libs` flag make sense as a name?
</blockquote></details>

I'm not sure of why we'd need to specify multiple directories. I think we can start with a single directory. Should this be geo-specific, or general? `--libs` or `--geo-libs`? My instinct is for `--libs`, though I don't have any logical argument for that. I was also imagining this to be the directory name, and not the library path name, but perhaps that is more error prone as then we're implicitly requiring a specific filename within the directory. I'm going to again invoke @rmloveland to give his opinion from a user/docs perspective.
___
*[pkg/geo/geos/geos_unix.go, line 87 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-M5SJXaEBWC08EFx_KW4:-M5YVkls8OEexORCrNZQ:bj8l2j7) ([raw file](https://github.com/cockroachdb/cockroach/blob/8124961cde1a51bb9531a0159f04e210807556bb/pkg/geo/geos/geos_unix.go#L87)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

https://www.cockroachlabs.com/docs/stable/install-cockroachdb-mac.html
https://www.cockroachlabs.com/docs/stable/install-cockroachdb-linux.html

We tell the users to put it straight into /usr/local/bin.

> Why does /usr/local/lib get special treatment here? 

factory default for everything I guess. I'm happy to omit having `/usr/local/lib` as the default for now and we can have that discussion next meeting (if we can't decide in person).
</blockquote></details>

Rather than hardcoding a value that is always checked, perhaps we should make `/usr/local/lib` the default for a `--libs` flag. I think it would be quite surprising for `cockroach start --libs=foo` to succeed by finding libgeos in `/usr/local/lib` because the libgeos in `foo` is misnamed.

**Review by rmloveland — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47759)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz, @petermattis, and @rmloveland)

___
*[pkg/geo/geos/geos_unix.go, line 87 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-M5SJXaEBWC08EFx_KW4:-M5YWKs43JCjFcys8NOp:bgsmmqj) ([raw file](https://github.com/cockroachdb/cockroach/blob/8124961cde1a51bb9531a0159f04e210807556bb/pkg/geo/geos/geos_unix.go#L87)):*
> perhaps we should make /usr/local/lib the default for a --libs flag

That would also work for me

**Comment by otan** [comment]
___
*[pkg/cli/cliflags/flags.go, line 1054 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-M5TeQbQ9sCLYTbBIftO:-M5Yffxl4KIX8zxX7O6A:b-t8iwpc) ([raw file](https://github.com/cockroachdb/cockroach/blob/620a517e6ca554652459907f7983b17ca7a734d5/pkg/cli/cliflags/flags.go#L1054)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

I'm not sure of why we'd need to specify multiple directories. I think we can start with a single directory. Should this be geo-specific, or general? `--libs` or `--geo-libs`? My instinct is for `--libs`, though I don't have any logical argument for that. I was also imagining this to be the directory name, and not the library path name, but perhaps that is more error prone as then we're implicitly requiring a specific filename within the directory. I'm going to again invoke @rmloveland to give his opinion from a user/docs perspective.
</blockquote></details>

my only problem with it is that if the name of the extension is different (e.g. libgeos_c.a) then we won't catch it.

**Comment by otan** [comment]
___
*[pkg/cli/cliflags/flags.go, line 1054 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-M5TeQbQ9sCLYTbBIftO:-M5Yfqcc9BoUASqpLMrN:b-kzo7go) ([raw file](https://github.com/cockroachdb/cockroach/blob/620a517e6ca554652459907f7983b17ca7a734d5/pkg/cli/cliflags/flags.go#L1054)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

my only problem with it is that if the name of the extension is different (e.g. libgeos_c.a) then we won't catch it.
</blockquote></details>

(a directory i mean)

**Comment by otan** [comment]
___
*[pkg/geo/geos/geos_unix.go, line 87 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-M5SJXaEBWC08EFx_KW4:-M5YgAHt77cTL9uQpM-x:bwqzc58) ([raw file](https://github.com/cockroachdb/cockroach/blob/8124961cde1a51bb9531a0159f04e210807556bb/pkg/geo/geos/geos_unix.go#L87)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

> perhaps we should make /usr/local/lib the default for a --libs flag

That would also work for me

</blockquote></details>

(note that we always take the flag as source of truth for now, but also happy for that to be the default)

**Review by petermattis — APPROVED** [review]
:lgtm: modulo one more small comment about the `--geos-libs` help text. Also, the commit message says `--geos-lib` in the body, while it should say `--geos-libs`.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-:-M5cjhhqB2Cg02FV8u1f:br4wy70)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @knz, @otan, and @rmloveland)

___
*[pkg/cli/cliflags/flags.go, line 1057 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/47759#-M5cjSie6XQoU6FlmAmi:-M5cjSie6XQoU6FlmAmj:b-lmz62x) ([raw file](https://github.com/cockroachdb/cockroach/blob/3af1d9dd668c6dadfc9d3361b5b3eed14355ba4a/pkg/cli/cliflags/flags.go#L1057)):*
> ```Go
> 		Description: `
> The location where all libraries for Geospatial operations is located.
> Defaults to /usr/local/lib`,
> ```

I think `cobra.Command` (or something) automatically prints the default.

**Comment by otan** [comment]
tftrs everyone!

bors r=petermattis
